apps/homescreen/js/pages.js

```diff
@@ -219,14 +219,11 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
-      this.pagesStore.get(id).then(entry => {
-        entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
-          e => {
-            console.error('Error unpinning page:', e);
-          });
-      }, e => {
-        console.error('Error retrieving page to unpin:', e);
+      // TODO: fix LazyLoader mock to support promises
+      return new Promise((resolve, reject) => {
+        LazyLoader.load(['shared/js/places_model.js'], () => {
```

Comment:
This is true - but then, we already assume all of that in the other code that accesses this datastore. If all places access was replaced, I'd agree, but as it is, it's just one small part of it that could easily just be here and then provide a good example for people looking for datastore examples. I'd still prefer that check to exist in homescreen, or alternatively, for all of the homescreen places access to be replaced by the shared code (and for that shared code to be at least as efficient as what it's replacing).